src/login/logind-session.c

```diff
@@ -959,7 +961,6 @@ int session_stop(Session *s, bool force) {
                 return 0;
 
         s->timer_event_source = sd_event_source_unref(s->timer_event_source);
-        s->leader_pidfd_event_source = sd_event_source_unref(s->leader_pidfd_event_source);
```

Comment:
this line should mvoe to session_dispatch_leader_pidfd() then so that we don't keep the pidfd open forever once we saw the process being killed.

Reply:
Done.